nwcsafpps_runner/tests/test_utils.py

```diff
@@ -23,14 +23,66 @@
 import unittest
 import pytest
 from datetime import datetime
-
+from nwcsafpps_runner.utils import create_xml_timestat_from_lvl1c, create_product_statistics_from_lvl1c
 from nwcsafpps_runner.utils import get_outputfiles
 from nwcsafpps_runner.utils import get_time_control_ascii_filename_candidates
 from nwcsafpps_runner.utils import get_time_control_ascii_filename
 from nwcsafpps_runner.utils import FindTimeControlFileError
 from nwcsafpps_runner.utils import get_product_statistics_files
 
 
+def test_create_xml_timestat_from_lvl1c(tmp_path):
+    """Test xml files that for a level1c file."""
+    #: Create temp_path
+    mydir = tmp_path / "import"
+    mydir.mkdir()
+    mydir_out = tmp_path / "export"
+    mydir_out.mkdir()
+
+    #: Create test files
+    def create_files(mydir, file_tag, typ):
+        #: These files should always be found
+        f1 = mydir / "S_NWC_{}_npp_12345_19810305T0715000Z_19810305T0730000Z{}".format(file_tag, typ)
+        f1.write_text("test_file type {:s}".format(typ))
+        f1 = mydir / "S_NWC_{}_noaa20_12345_19810305T0715000Z_19810305T0730000Z{}".format(file_tag, typ)
+        f1.write_text("test_file type {:s}".format(typ))
+        f1 = mydir / "S_NWC_{}_npp_82345_19820305T0715000Z_19820305T0730000Z{}".format(file_tag, typ)
+        f1.write_text("test_file type {:s}".format(typ))
+
+    #: Create the level1c and xml files
+    create_files(mydir, "viirs", ".nc")
+    create_files(mydir_out, "CMAPROB", ".nc")
+    create_files(mydir_out, "CMAPROB", "_statistics.xml")
+    create_files(mydir_out, "CMIC", "_statistics.xml")
+    create_files(mydir_out, "CTTH", "_statistics.xml")
+    create_files(mydir_out, "timectrl", ".xml")
+    create_files(mydir_out, "timectrl", "_dummy.xml")
+    scene = {'file4pps': "S_NWC_viirs_npp_12345_19810305T0715000Z_19810305T0730000Z.nc"}
+
+    #: Test xml files for timectrl
+    res = create_xml_timestat_from_lvl1c(scene, mydir_out)
+    expected = [os.path.join(mydir_out, "S_NWC_timectrl_npp_12345_19810305T0715000Z_19810305T0730000Z.xml")]
+    assert len(res) == len(set(res))
+    assert set(res) == set(expected)
+
+    #: Test xml files for products
+    res = create_product_statistics_from_lvl1c(scene, mydir_out)
+    expected = [os.path.join(mydir_out, "S_NWC_CMAPROB_npp_12345_19810305T0715000Z_19810305T0730000Z_statistics.xml"),
+                os.path.join(mydir_out, "S_NWC_CTTH_npp_12345_19810305T0715000Z_19810305T0730000Z_statistics.xml"),
+                os.path.join(mydir_out, "S_NWC_CMIC_npp_12345_19810305T0715000Z_19810305T0730000Z_statistics.xml")]
+    assert len(res) == len(set(res))
+    assert set(res) == set(expected)
+
+    scene = {}
+    #: Test xml files for timectrl no file4pps attribute
+    res = create_xml_timestat_from_lvl1c(scene, mydir_out)
+    assert res == []
+
+    #: Test xml files for products no file4pps attribute
+    res = create_product_statistics_from_lvl1c(scene, mydir_out)
+    assert res == []
```

Comment:
These look like 4 different tests, could they be split into different functions? Maybe into a class, or with a new fixture for the file creations?

Reply:
I followed the strategy of test_outputfiles and kept them in one function to just create files ones. I read here https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html. I did not understand a way to use the tmp_path with the unittest.TestCase and setUp/tearDown. Do you know of an example where tmp_path is used in better way?

Reply:
Here is an example with a fixture:
https://github.com/pytroll/satpy/blob/324bf13dd63174db8bc2efeb0364660807b9cd23/satpy/tests/reader_tests/test_oceancolorcci_l3_nc.py#L93-L125

Reply:
and here an example with `setup_method` https://github.com/pytroll/satpy/blob/37e77dd2921c882dd501c68fc2837942dbe51fe6/satpy/tests/reader_tests/test_ghrsst_l2.py#L35-L73